pulp_docker/app/models.py

```diff
@@ -1,37 +1,177 @@
+    MANIFEST_LIST='application/vnd.docker.distribution.manifest.list.v2+json',
+    CONFIG_BLOB='application/vnd.docker.container.image.v1+json',
+    REGULAR_BLOB='application/vnd.docker.image.rootfs.diff.tar.gzip',
+    FOREIGN_BLOB='application/vnd.docker.image.rootfs.foreign.diff.tar.gzip',
+)
+
+
+class Manifest(Content):
+    """
+    A docker manifest.
+
+    This content has no artifacts.
+
+    Fields:
+        digest (models.CharField): The manifest digest.
+        schema_version (models.IntegerField): The docker schema version.
+        media_type (models.CharField): The manifest media type.
     """
-    The "docker" content type.
 
-    Define fields you need for your new content type and
-    specify uniqueness constraint to identify unit of this type.
+    TYPE = 'manifest'
+
+    digest = models.CharField(max_length=255)
+    schema_version = models.IntegerField()
+    media_type = models.CharField(
+        max_length=60,
+        choices=(
+            (MEDIA_TYPE.MANIFEST_V1, MEDIA_TYPE.MANIFEST_V1),
+            (MEDIA_TYPE.MANIFEST_V2, MEDIA_TYPE.MANIFEST_V2),
+        ))
 
-    For example::
+    class Meta:
+        unique_together = ('digest',)
 
-        field1 = models.TextField()
-        field2 = models.IntegerField()
-        field3 = models.CharField()
 
-        class Meta:
-            unique_together = (field1, field2)
+class ManifestBlob(Content):
     """
+    A blob defined within a manifest.
 
-    TYPE = 'docker'
+    The actual blob file is stored as an artifact.
+
+    Fields:
+        digest (models.CharField): The blob digest.
+        media_type (models.CharField): The blob media type.
+
+    Relations:
+        manifest (models.ForeignKey): Many-to-one relationship with Manifest.
+    """
+
+    TYPE = 'manifest-blob'
+
+    digest = models.CharField(max_length=255)
+    media_type = models.CharField(
+        max_length=80,
+        choices=(
+            (MEDIA_TYPE.CONFIG_BLOB, MEDIA_TYPE.CONFIG_BLOB),
+            (MEDIA_TYPE.REGULAR_BLOB, MEDIA_TYPE.REGULAR_BLOB),
+            (MEDIA_TYPE.FOREIGN_BLOB, MEDIA_TYPE.FOREIGN_BLOB),
+        ))
+
+    manifest = models.ForeignKey(Manifest, related_name='blobs', on_delete=models.CASCADE)
+
+    class Meta:
+        unique_together = ('digest',)
+
+
+class ManifestList(Content):
+    """
+    A blob defined within a manifest.
+
+    This content has no artifacts.
+
+    Fields:
+        digest (models.CharField): The manifest digest.
+        schema_version (models.IntegerField): The docker schema version.
+        media_type (models.CharField): The manifest media type.
+
+    Relations:
+        manifests (models.ManyToManyField): Many-to-many relationship with Manifest.
+    """
+
+    TYPE = 'manifest-list'
+
+    digest = models.CharField(max_length=255)
+    schema_version = models.IntegerField()
+    media_type = models.CharField(
+        max_length=60,
+        choices=(
+            (MEDIA_TYPE.MANIFEST_LIST, MEDIA_TYPE.MANIFEST_LIST),
```

Comment:
These 2 choices are the same. I think there is only one choice, maybe this one doesn't need to be a choice field.

Reply:
that's true that there is just 1 choice  - for now ( you never know what will come next)
+ the value of this field will be validated at the db level.

Reply:
Just noticed that this is (visible choice, internal choice), so this is just 1 choice. It seems reasonable to expect docker to add more choices in the future so this is fine.